hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/cli/GenericCli.java

```diff
@@ -123,4 +132,25 @@ protected PrintWriter out() {
   protected PrintWriter err() {
     return cmd.getErr();
   }
+
+  private String handleFileSystemException(FileSystemException e) {
+    // If reason is set, return the exception's message as it is.
+    if (e.getReason() != null) {
+      return e.getMessage();
+    }
+
+    // Otherwise, construct a custom message based on the type of exception
+    String errorMessage;
+    if (e instanceof NoSuchFileException) {
+      errorMessage = String.format("Error: File not found: %s", e.getFile());
+    } else if (e instanceof AccessDeniedException) {
+      errorMessage = String.format("Error: Access denied to file: %s", e.getFile());
+    } else if (e instanceof FileAlreadyExistsException) {
+      errorMessage = String.format("Error: File already exists: %s", e.getFile());
+    } else {
+      errorMessage = String.format("Error with file: %s. Details: %s", e.getFile(), e.getMessage());
```

Comment:
Include exception type in the message, but can omit file, since it's supposed to be included in `getMessage()`.